.ci/test.py

```diff
@@ -306,20 +311,82 @@ def run_all(args):
     root = Path(root_path)
 
     plugins = list(enumerate_plugins(root))
-    if args != []:
-        plugins = [p for p in plugins if p.name in args]
+    if plugin_names != []:
+        plugins = [p for p in plugins if p.name in plugin_names]
         print("Testing the following plugins: {names}".format(names=[p.name for p in plugins]))
     else:
         print("Testing all plugins in {root}".format(root=root))
 
     results = [(p, run_one(p)) for p in plugins]
     success = all([t[1] for t in results])
 
+    if sys.version_info[0:2] == (3, 12) and update_badges:
```

Comment:
We discussed moving this step to the end of the action so that the badge can be red if any *one* of the python versions fails, rather than only marking what 3.12 did

Reply:
Updating the badges data (json files) has now been moved to the `gather` step in the `main` and `nightly` workflows. A green badge now requires all Python versions to have had successful runs. To this end a new file `update_badges.py` which contains the update logic.